src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1047,6 +1031,7 @@ namespace cryptonote
   void core::on_synchronized()
   {
     m_miner.on_synchronized();
+    m_blockchain_storage.safesyncmode();
```

Comment:
The checks before calling on_connection_synchronized() already compare current height to target_blockchain_height.